contracts/contracts/dapphack.sol

```diff
@@ -134,9 +139,51 @@ contract DappHack is ProjectNFTs {
     }
 
     //create mapping for this
-    modifier NotInTeam() {
+    modifier NotInTeam(address[] memory participants) {
         // need a mapping from address(of the signer) to team as cant fetch a team that is not created
-        // require(s_teams[msg.sender] == 0, "Already in a team");
+
+        for (uint i = 0; i < participants.length; i++) {
+            require(
+                bytes(builderToTeam[participants[i]].name).length == 0,
+                "Already in a team"
+            );
+        }
+
+        require(
+            bytes(builderToTeam[msg.sender].name).length == 0,
+            "Already in a team"
+        );
+        _;
+    }
+
+   modifier DuplicateParticipants(address[] memory participant) {
```

Comment:
Hey this is quite inefficient, you can do something like this 
```
modifier DuplicateParticipants(address[] memory participant) {
    mapping(address => bool) seen;
    for (uint i = 0; i < participant.length; i++) {
        if (seen[participant[i]]) {
            revert("Duplicate Participants");
        }
        seen[participant[i]] = true;
    }
    // Check msg.sender separately if needed
    if (seen[msg.sender]) {
        revert("Duplicate Participants including msg.sender");
    }
    _;
}
```

Reply:
![image](https://github.com/BlocSoc-iitr/DappHack/assets/136801346/6a595a7a-6572-4481-8e72-066fcb483760)
![image](https://github.com/BlocSoc-iitr/DappHack/assets/136801346/fa818f0d-928d-4b72-9cb6-6e57d15bce01)
Ye error aa rhe hai sir

Reply:
Ahh sorry, mappings  can't be declared dynamically in modifier, hmm, then do something simple like this
```
    modifier DuplicateParticipants(address[] memory participants) {
        for (uint i = 0; i < participants.length; i++) {
            for (uint j = i + 1; j < participants.length; j++) {
                if (participants[i] == participants[j]) {
                    revert("Duplicate Participants");
                }
            }
        }
        _;
    }
    ```
    and handle the msg.sender case as well if not included in participants.

Reply:
completed the changes